From 56be93ca1018bc16f94ead6fab67ae43eb26ee4c Mon Sep 17 00:00:00 2001
From: Dan Douglass <dan.douglass@freescale.com>
Date: Wed, 20 Jan 2016 13:31:56 -0600
Subject: [PATCH 0900/5242] MLK-12302 caam: Secure Memory platform device
 creation crashes

commit  7728199bf175ea12dc186e2f7ba14571d9d5209c from
https://source.codeaurora.org/external/imx/linux-imx.git

There is new flag that is checked in of_platform_device_create().
The flag is cleared prior to the call now so the device creates
successfully.

Signed-off-by: Dan Douglass <dan.douglass@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/crypto/caam/sm_store.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/crypto/caam/sm_store.c b/drivers/crypto/caam/sm_store.c
index ab676be..a52a854 100644
--- a/drivers/crypto/caam/sm_store.c
+++ b/drivers/crypto/caam/sm_store.c
@@ -36,6 +36,7 @@
 #include "intern.h"
 #include "error.h"
 #include "sm.h"
+#include <linux/of_address.h>
 
 #define SECMEM_KEYMOD_LEN 8
 #define GENMEM_KEYMOD_LEN 16
@@ -969,6 +970,8 @@ int caam_sm_startup(struct platform_device *pdev)
 	/* Create the dev */
 #ifdef CONFIG_OF
 	np = of_find_compatible_node(NULL, NULL, "fsl,imx6q-caam-sm");
+	if (np)
+		of_node_clear_flag(np, OF_POPULATED);
 	sm_pdev = of_platform_device_create(np, "caam_sm", ctrldev);
 #else
 	sm_pdev = platform_device_register_data(ctrldev, "caam_sm", 0,
-- 
1.7.9.5

